### PR TITLE
Improve error handling

### PR DESCRIPTION
**Pull request by georgolden**
- [ ] tests and linter show no problems (`npm t`)
- [ ] tests are added/updated for bug fixes and new features
- [x] code is properly formatted (`npm run fmt`)
- [ ] description of changes is added in CHANGELOG.md

### CODE REVIEW

**Comment by georgolden** [comment]
https://github.com/metarhia/impress/issues/1689

File: metalog.js [hunk 1]
```diff
@@ -291,11 +293,16 @@ class Logger extends events.EventEmitter {
...
           fs.stat(fileName, (err, stats) => {
-            if (!err && stats.size === 0) {
-              fsp.unlink(this.file);
+            if (err) {
+              reject(err);
+              return;
+            }
+            if (stats.size && stats.size === 0) {
+              fsp.unlink(fileName).then(resolve).catch(reject);
+              return;
```
Comment (tshemsedinov) [inline metalog.js:303]: We don not need to reject here. This `fs.stat` is just to check if file exists

File: metalog.js [hunk 2]
```diff
@@ -291,11 +293,16 @@ class Logger extends events.EventEmitter {
...
           const fileName = this.file;
           this.emit('close');
-          resolve();
           fs.stat(fileName, (err, stats) => {
-            if (!err && stats.size === 0) {
-              fsp.unlink(this.file);
+            if (err) {
+              reject(err);
+              return;
+            }
```
Comment (tshemsedinov) [inline metalog.js:300]:
```suggestion
            if (err) return;
```

**Review by tshemsedinov — APPROVED** [review]
Going to land this without package-lock changes

File: metalog.js [hunk 3]
```diff
@@ -291,11 +293,13 @@ class Logger extends events.EventEmitter {
           this.reopenTimer = null;
           const fileName = this.file;
           this.emit('close');
-          resolve();
           fs.stat(fileName, (err, stats) => {
-            if (!err && stats.size === 0) {
-              fsp.unlink(this.file);
+            if (err) return;
+            if (stats.size && stats.size === 0) {
+              fsp.unlink(fileName).then(resolve).catch(reject);
```
Comment (tshemsedinov) [inline metalog.js:299]:
```suggestion
            if (!err && stats.size && stats.size === 0) {
              fsp.unlink(fileName).catch(() => {});
```

File: metalog.js [hunk 4]
```diff
@@ -291,11 +293,13 @@ class Logger extends events.EventEmitter {
...
+              return;
```
Comment (tshemsedinov) [inline metalog.js:300]:
```suggestion
```

File: metalog.js [hunk 5]
```diff
@@ -291,11 +293,11 @@ class Logger extends events.EventEmitter {
           this.reopenTimer = null;
           const fileName = this.file;
           this.emit('close');
-          resolve();
           fs.stat(fileName, (err, stats) => {
-            if (!err && stats.size === 0) {
-              fsp.unlink(this.file);
+            if (!err && stats.size && stats.size === 0) {
```
Comment (tshemsedinov) [inline metalog.js:297]:
```suggestion
            if (!err && stats.size === 0) {
```

**Comment by tshemsedinov** [comment]
Landed in https://github.com/metarhia/metalog/commit/edc95f710f48bd5212b52b8ab733f7d221fd4382

**Pull request closed without merging**